packages/jest-haste-map/src/getMockName.js

```diff
@@ -0,0 +1,13 @@
+const path = require('path');
+
+const getMockName = (filePath, mocksPattern) => {
+  const modulePath = filePath.split(mocksPattern)[1];
+
+  const extractMockName = new RegExp(
+    `^\\${path.sep}?(.*?)(\\${path.sep}index)?\\${path.extname(modulePath)}$`
+  );
```

Comment:
I'm really worried about performance here. Right now, mocksPattern is a config option but it isn't *really* configurable and I'm ok making it non-configurable just like the snapshots folder because it causes trouble.

Can you change this either by:
* Optimize this code for performance. Think running this on 50k files, it should be less than 100ms or at least not slower than what we do here currently.
* Change the mocksPattern option to be hard-coded, so splitting at `__mocks__` and looking at the right-hand side of this should work and be fast.